classes/CountryStateDynamicMap.cls

```diff
@@ -0,0 +1,105 @@
+	public CountryStateDynamicMap(String keyword) {
+        List<District__c> listOfDistrict =  [SELECT Name, State_Name__r.Name ,
+                                             State_Name__r.Country_Name__r.Name FROM District__c];
+        //System.debug(listOfDistrict);                   
+        for(District__c dis : listOfDistrict) {
+            if(!mapCountryTillDistricts.containsKey(dis.State_Name__r.Country_Name__r.Name)){
+                mapCountryTillDistricts.put(dis.State_Name__r.Country_Name__r.Name, new Map<String,List<String>>());
+            }
+            Map<String,List<String>> MapOfState = mapCountryTillDistricts.get(dis.State_Name__r.Country_Name__r.Name);
+            if(!MapOfState.containsKey(dis.State_Name__r.Name)){
+                MapOfState.put(dis.State_Name__r.Name, new List<String>());
+            }
+            MapOfState.get(dis.State_Name__r.Name).add(dis.Name);
+        }
+        // System.debug(mapCountryTillDistricts.get('India'));
+        //1.finding combination of Country, state and districts where it starts with 'A'
+        System.debug('Comibinations of Countries, States and Districts which starts with A : ');
+        for(String getCountry :  mapCountryTillDistricts.keyset()) {
+            if(getCountry.startsWithIgnoreCase('A')) {
+                System.debug('Country - '+getCountry);
+                for(String getState : mapCountryTillDistricts.get(getCountry).keySet()) {
+                    if(getState.startsWIthIgnoreCase('A')) {
+                        System.debug('State - '+getState);
+                        for(String getDistrict : mapCountryTillDistricts.get(getCountry).get(getState)) {
+                            if(getDistrict.startsWIthIgnoreCase('A')) {
+                                System.debug('District - '+getDistrict);
+                            }
+                        }
+                    }  
+                }
+            }
+        }
+        // 2. Find such a District which length is more or equal to some number.
+        System.debug('District which length is more or equal to 6 : ');
+        for(String getCountry : mapCountryTillDistricts.keySet()) {
+            for(String getState : mapCountryTillDistricts.get(getCountry).keySet()) {
+                for(String getDistrict : mapCountryTillDistricts.get(getCountry).get(getState)) {
+                    if(getDistrict.length() <= 6) {
+                        System.debug('District : '+getDistrict);
+                    }
+                }
+            }
+        }
+    }     
+    //JSON  wrapper classes
+    public List<Country> countryList{get; set;}
+    public class Districts {
+        public String districtName{get; set;}
+    }        
+    public class States{
+        public string stateName{get; set;}
+        public List<Districts> districts{get; set;}
+        public Integer Total_Number_Of_Districts{get; set;}
+    }
+    Integer Stdata ;
```

Comment:
what is stData?  there should not be any method after the wrapper class

Reply:
that was a variable that defined for calculating the total number of district on state boject, i forggot to remove it,without this variable, we can  populate the total number of the distict on state object .